packages/ui/dialogs/remove-account/index.tsx

```diff
@@ -15,32 +20,42 @@ const {
 } = language
 const removeAccountText = language.removeAccount
 
-import Button from 'ui/components/button'
-
 import style from 'theme/form'
 import general from 'theme/general'
 
+const removingAccount = new LoadingContext()
+
 interface Props {
   removeAccount: () => any
-  cancelRemoveAccount: () => any
+  hasStoredUser: boolean
+  navigation: any
 }
 
 class RemoveAccountView extends Component<Props> {
+  constructor(props) {
+    super(props)
+
+    this.removeAccount = this.removeAccount.bind(this)
+  }
+
+  async removeAccount() {
+    await removingAccount.wrap(this.props.removeAccount())
+    this.props.navigation.dispatch(getResetAction( { routeName: 'Dashboard' } ))
```

Comment:
I may not be following this correctly, so correct me if I'm wrong.

Line 42 will always resolve, even if account removal fails (it just displays an error but still resolves), meaning that line 43 will execute and the user will be navigated back to the dashboard. This *could* mislead the user into thinking the account removal succeeded, if say, they don't see the error notification or don't infer its relevance to the account removal.

The logic for navigating to the dashboard should, then, occur only if removal succeeds. The only place that currently knows of this is the account removal action itself... but one could argue that that action shouldn't be concerned with app navigation. Finding the right place for that logic, I'll leave to your judgment. As long as account removal fails in an obvious way and doesn't navigate the user away.

Again, I may be misinterpreting how the `removeAccount()` prop is wired. Then again, if it *does* reject on failure, we still have a problem, because that would cause it to throw an uncaught promise rejection within a React component, which can "break the app" and cause the UI to become unresponsive (or, at least, React components used to fail loudly on this sort of thing).